zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/Archiver.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.archiver;
+
+import io.camunda.exporter.config.ExporterConfiguration.ArchiverProperties;
+import io.camunda.exporter.metrics.CamundaExporterMetrics;
+import io.camunda.zeebe.util.CloseableSilently;
+import io.camunda.zeebe.util.VisibleForTesting;
+import io.camunda.zeebe.util.error.FatalErrorHandler;
+import java.util.Objects;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import javax.annotation.WillCloseWhenClosed;
+import org.agrona.CloseHelper;
+import org.slf4j.Logger;
+
+@SuppressWarnings({"unused", "FieldCanBeLocal"}) // can be removed in the future
+public final class Archiver implements CloseableSilently {
+  private final int partitionId;
+  private final ArchiverRepository repository;
+  private final ArchiverProperties properties;
+  private final CamundaExporterMetrics metrics;
+  private final Logger logger;
+  private final ScheduledExecutorService executor;
+
+  @VisibleForTesting
+  Archiver(
+      final int partitionId,
+      final @WillCloseWhenClosed ArchiverRepository repository,
+      final ArchiverProperties properties,
+      final CamundaExporterMetrics metrics,
+      final Logger logger,
+      final @WillCloseWhenClosed ScheduledExecutorService executor) {
+    this.partitionId = partitionId;
+    this.repository = Objects.requireNonNull(repository, "must specify a repository");
+    this.properties = Objects.requireNonNull(properties, "must specify properties");
+    this.metrics = Objects.requireNonNull(metrics, "must specify metrics");
+    this.logger = Objects.requireNonNull(logger, "must specify a logger");
+    this.executor = Objects.requireNonNull(executor, "must specify an executor");
+  }
+
+  @Override
+  public void close() {
+    // avoid calling executor.close, which will await 1d (!) until termination
+    // we also don't need to wait for the jobs to fully finish, as we should be able to handle
+    // partial jobs (e.g. node crash/restart)
```

Comment:
Hm, good point. If I close the repository and a task was already submitted, it may try to use it. We will likely need some kind of synchronization in the repository to avoid access once closed, and also change the error handling in the jobs (e.g. if the repository closed the underlying client while we were waiting for a result, but the archiver is closed, we can ignore the error).